### PR TITLE
chore(): index files

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

index files for exporting and tree shaking

## Changes

- index files
- default controls refactor: split into a file for exports and a file for side effects

### CODE REVIEW

**Comment by ShaMan123** [comment]
I will remove the test to a dedicated PR

**Review by ShaMan123 — COMMENTED** [review]
I would like to move classREgistry from util to src and out of the utils export scope since it doesn't belong there, it is not a util
And I want to rename controlsUtils to controls

Awaiting Green light

**Review by ShaMan123 — COMMENTED** [review]
ready

File: fabric.ts [hunk 1]
```diff
@@ -1,425 +1,57 @@
...
-  getWindow,
-  setEnvForTests,
-  parseStyleAttribute,
-  parsePointsAttribute,
-  parseFontDeclaration,
-  parseTransformAttribute,
-  getCSSRules,
-};
+} from './src/filters';
+export * as filters from './src/filters';
```
Comment (asturur) [inline fabric.ts:57]: doesn't this re-export also the one we exported above?

File: fabric.ts [hunk 2]
```diff
@@ -1,425 +1,57 @@
...
+export { IText } from './src/shapes/IText/IText';
+export { Textbox } from './src/shapes/Textbox';
+export { Group } from './src/shapes/Group';
+export { ActiveSelection } from './src/shapes/ActiveSelection';
+export { Image } from './src/shapes/Image';
+export { createCollectionMixin } from './src/Collection';
+
+export * as util from './src/util';
+
+export * from './src/parser';
```
Comment (asturur) [inline fabric.ts:45]: Not being a named export this seems just hides visibility. Can we see the named exports here?

File: fabric.ts [hunk 3]
```diff
@@ -1,425 +1,57 @@
...
+
+export { Control } from './src/controls';
+export * as controlsUtils from './src/controls';
```
Comment (asturur) [inline fabric.ts:48]: as for filters, isn't this a doulbe export? controlsUtils.Control and Control?

**Comment by asturur** [comment]
> I would like to move classREgistry from util to src and out of the utils export scope since it doesn't belong there, it is not a util And I want to rename controlsUtils to controls
> 
> Awaiting Green light

We can move classRegistry around, but control utils are controls utils, aren't controls per se.
Also

**Comment by asturur** [comment]
it still creates those things here:

```js
var index = /*#__PURE__*/Object.freeze({
  __proto__: null,
  BaseFilter: BaseFilter,
  BlackWhite: BlackWhite,
  BlendColor: BlendColor,
  BlendImage: BlendImage,
  Blur: Blur,
  Brightness: Brightness,
  Brownie: Brownie,
  Canvas2dFilterBackend: Canvas2dFilterBackend,
  ColorMatrix: ColorMatrix,
  Composed: Composed,
  Contrast: Contrast,
  Convolute: Convolute,
  Gamma: Gamma,
  Grayscale: Grayscale,
  HueRotation: HueRotation,
  Invert: Invert,
  Kodachrome: Kodachrome,
  Noise: Noise,
  Pixelate: Pixelate,
  Polaroid: Polaroid,
  RemoveColor: RemoveColor,
  Resize: Resize,
  Saturation: Saturation,
  Sepia: Sepia,
  Technicolor: Technicolor,
  Vibrance: Vibrance,
  Vintage: Vintage,
  WebGLFilterBackend: WebGLFilterBackend,
  getFilterBackend: getFilterBackend,
  initFilterBackend: initFilterBackend
});
```
So i m not sure how this is different from what we have now.

File: fabric.ts (hunk 1 above)
Comment (ShaMan123) [reply fabric.ts:57]: yes you are right
should the above belong to filters and be exported as standalone? I think it should

File: fabric.ts (hunk 2 above)
Comment (ShaMan123) [reply fabric.ts:45]: I don't mind but is missing the whole point of moving it to a standalone index
I can return them here

File: fabric.ts (hunk 3 above)
Comment (ShaMan123) [reply fabric.ts:48]: > controlsUtils.Control and Control?

This is part of why I want to rename so then `controls.Control` `controls.wrapWithXXX`
A nice scoped namespace

**Comment by ShaMan123** [comment]
For direct import it makes a difference for sure
FYI: `/*#__PURE__*/` this annotation is described in rollup tree shaking to instruct the bundler not to treat it as a side effect

File: fabric.ts (hunk 2 above)
Comment (ShaMan123) [reply fabric.ts:46]: I want them to be
`export * as parser from './src/parser';`

**Review by ShaMan123 — COMMENTED** [review]
renamed `cotrols.render` => `controlRendering`
renamed and move class registry

**Pull request merged**